server/datastore/mysql/schema.sql

```diff
@@ -119,8 +119,7 @@ CREATE TABLE `enroll_secrets` (
 CREATE TABLE `host_additional` (
   `host_id` int(10) unsigned NOT NULL,
   `additional` json DEFAULT NULL,
-  PRIMARY KEY (`host_id`),
-  CONSTRAINT `host_additional_ibfk_1` FOREIGN KEY (`host_id`) REFERENCES `hosts` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
One thing to keep in mind is: 

The foreign key references in `host_additional`, `policy_membership` and `host_users` were useful for one scenario:

0. Fleet provides policy/additional/host-users queries to hosts (`distributed/read`).
1. User or cron deletes host X.
2. Host X check-in with the results for queries provided in 0 (`distributed/write`).

With the changes added here, there will be e.g. `policy_membership` entries for a host that doesn't exist anymore (previously, the check-in would have failed due to foreign key reference not existing).

Reply:
Though I would have to double check, maybe this is protected by the fact that loading the host to authenticate it in the `distributed/write` will fail early?

Reply:
Right, a deleted host won't be able to be authenticated.

Reply:
Another reason to give more thought to caching `AuthenticateHost`...